pkg/util/labels.go

```diff
@@ -101,6 +101,81 @@ func AddObjectLabels(obj runtime.Object, labels labels.Set) error {
 	return nil
 }
 
+// AddObjectAnnotations adds new annotation(s) to a single runtime.Object
+func AddObjectAnnotations(obj runtime.Object, annotations map[string]string) error {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	accessor, err := kmeta.Accessor(obj)
+
+	if err != nil {
+		if _, ok := obj.(*runtime.Unstructured); !ok {
+			// error out if it's not possible to get an accessor and it's also not an unstructured object
+			return err
+		}
+	} else {
+		metaAnnotations := accessor.Annotations()
+		if metaAnnotations == nil {
+			metaAnnotations = make(map[string]string)
+		}
+
+		switch objType := obj.(type) {
+		case *deployapi.DeploymentConfig:
+			if err := addDeploymentConfigNestedAnnotations(objType, annotations); err != nil {
+				return fmt.Errorf("unable to add nested annotations to %s/%s: %v", accessor.Kind(), accessor.Name(), err)
+			}
+		}
+
+		if err := MergeInto(metaAnnotations, annotations, ErrorOnDifferentDstKeyValue); err != nil {
```

Comment:
I would not throw an error on a different dstkeyvalue... if you are generating new objects based on existing objects generated by a different generator, you'd likely still want to set the new generated-by annotation.